lib/alerts_viewer_web/live/alerts_to_close_live.ex

```diff
@@ -0,0 +1,73 @@
+defmodule AlertsViewerWeb.AlertsToCloseLive do
+  @moduledoc """
+  LiveView for displaying alerts that we can maybe close.
+  """
+  use AlertsViewerWeb, :live_view
+  alias Alerts.Alert
+  alias Routes.{Route, RouteStats, RouteStatsPubSub}
+  alias TripUpdates.TripUpdatesPubSub
+
+  @impl true
+  def mount(_params, _session, socket) do
+    bus_alerts =
+      if(connected?(socket),
+        do: Alerts.subscribe(),
+        else: []
+      )
+
+    stats_by_route = if(connected?(socket), do: RouteStatsPubSub.subscribe(), else: %{})
+
+    alerts_by_route = alerts_by_route(bus_alerts)
+
+    block_waivered_routes = if(connected?(socket), do: TripUpdatesPubSub.subscribe(), else: [])
+
+    socket =
+      assign(socket,
+        stats_by_route: stats_by_route,
+        block_waivered_routes: block_waivered_routes,
+        alerts_by_route: alerts_by_route
+      )
+
+    {:ok, socket}
+  end
+
+  @impl true
+
+  def handle_info({:alerts, alerts}, socket) do
+    alerts_by_route = alerts_by_route(alerts)
+    {:noreply, assign(socket, alerts_by_route: alerts_by_route)}
+  end
+
+  @impl true
+  def handle_info({:stats_by_route, stats_by_route}, socket) do
+    {:noreply, assign(socket, stats_by_route: stats_by_route)}
+  end
+
+  @impl true
+  def handle_info({:block_waivered_routes, block_waivered_routes}, socket) do
+    {:noreply, assign(socket, block_waivered_routes: block_waivered_routes)}
+  end
+
+  defp alerts_by_route(alerts) do
+    alerts
+    |> filtered_by_bus()
+    |> filtered_by_delay_type()
+    |> Alerts.by_route()
+    |> Map.to_list()
```

Comment:
I'm turning this into a keyword list here so it can be sorted by alert duration. It makes the keys a little awkward, since they have to be turned into atoms to make it a valid keyword list, but it was still the simplest solution I could think of.

Reply:
nitpick: I don't think you need the `Map.to_list()` line: you can call `Enum.map` on a `map`. ["Working with maps"](https://joyofelixir.com/10-maps) is a great page I look at all the time with a bunch of good examples.

Reply:
question: If you returned a list of `{route_id, alerts}` tuples, would that save you the step of converting the Route ID to an atom? (But it also doesn't really matter to me.)